llvm/lib/Support/raw_ostream.cpp

```diff
@@ -894,21 +904,30 @@ void raw_fd_ostream::anchor() {}
 raw_fd_ostream &llvm::outs() {
   // Set buffer settings to model stdout behavior.
   std::error_code EC;
-#ifdef __MVS__
-  EC = enablezOSAutoConversion(STDOUT_FILENO);
-  assert(!EC);
-#endif
+
+  // On z/OS we need to enable auto conversion
+  static bool convertedStdout = false;
+  if (!convertedStdout) {
+    EC = enableAutoConversion(STDOUT_FILENO);
+    assert(!EC);
+    convertedStdout = true;
+  }
+
```

Comment:
```suggestion
  static std::error_code EC1 = enableAutoConversion(STDOUT_FILENO);
  assert(!EC1);
  (void) EC1;
```
This is a little simpler.  The assert will be done every time but that should be ok.  The additional line is there to avoid warnings about unused variables with no assert builds.